banno-enrollment-po/BANNO.ENROLLMENT.V1.POW

```diff
@@ -0,0 +1,902 @@
+  FMERROR="New HB Username Cannot Be Blank"
+ IF FMERROR="" THEN
+  DO
+   FMPERFORM REVISE PREFERENCE LOC PREFLOC(SELECTEDPREF) (CHECKPRIVS,0,FMERROR)
+    DO
+     SET HBUSERNAME TO UPPERCASE(NEWUSERNAME)
+    END
+  END
+ IF FMERROR<>"" THEN
+  CALL ERRORHANDLER
+ ELSE IF FMERROR="" THEN
+  DO
+   PREFUSER(SELECTEDPREF)=NEWUSERNAME
+   POPUPMESSAGE(0,"Username Successfully Updated")
+  END
+END [PROCEDURE]
+
+PROCEDURE GETPREFERENCERECS
+[* Collects needed data from each HB preference on current
+** account.
+*]
+ PREFCOUNT=0
+ FOR EACH PREFERENCE WITH (PREFERENCE:HBUSERNAME<>"" AND
+                           PREFERENCE:HBMODE=STDHOMEBANKING)
+  DO
+   IF PREFCOUNT<PREFMAX THEN
+    DO
+     PREFCOUNT=PREFCOUNT+1
+     PREFLOC(PREFCOUNT)=PREFERENCE:LOCATOR
+     PREFUSER(PREFCOUNT)=PREFERENCE:HBUSERNAME
+     PREFEMAILADDR(PREFCOUNT)=PREFERENCE:EMAILADDRESS
+     PREFPASSPROB(PREFCOUNT)=PREFERENCE:PWDRESETPROBATION
+     PREFXFERANYLOC(PREFCOUNT)=0
+     FOR EACH PREFERENCE ACCESS WITH (PREFERENCE ACCESS:ACCESSTYPE=3)
+      DO
+       PREFXFERANYLOC(PREFCOUNT)=PREFERENCE ACCESS:LOCATOR
+      END
+     UNTIL PREFXFERANYLOC(PREFCOUNT)<>0
+   END
+  END
+ IF PREFCOUNT>PREFMAX THEN
+  DO
+   POPUPMESSAGE(2,"Too many users, Update Manually")
+   TERMINATE
+  END
+END [PROCEDURE]
+
+PROCEDURE CHOOSEACTION
+[* Dialog that prompts user for desired action.
+*]
+ DIALOGSTART("Banno Enrollment",1500%,0)
+ DIALOGPROMPTLISTSTART("Choose Action:",ACTIONCHGPASS)
+ IF PREFCOUNT>=1 THEN
+  DIALOGPROMPTLISTOPTION(ACTIONCHGPASS,"Change Password")
+ IF PREFCOUNT=0 OR ALLOWMULT=TRUE THEN
+  DIALOGPROMPTLISTOPTION(ACTIONCREATEUSR,"Create User")
+ IF PREFCOUNT>=1 THEN
+  DIALOGPROMPTLISTOPTION(ACTIONEDITUSER,"Edit User")
+ IF ALLOWDELETE=TRUE AND PREFCOUNT>=1 THEN
+  DIALOGPROMPTLISTOPTION(ACTIONDELUSER,"Delete User")
+ IF ALLOWXACCT=TRUE AND PREFCOUNT>=1 THEN
+  DO
+   DIALOGPROMPTLISTOPTION(ACTIONADDXACCT,"Add Cross Account")
+   DIALOGPROMPTLISTOPTION(ACTIONEDITXACCT,"Edit Cross Account")
+   DIALOGPROMPTLISTOPTION(ACTIONDELXACCT,"Delete Cross Account")
+  END
+ DIALOGPROMPTLISTOPTION(ACTIONEXITPGM,"Exit Banno Enrollment")
+ DIALOGPROMPTLISTEND
+ DIALOGDISPLAY
+ ACTIONREQUESTED=ENTERNUMBER("Choose Action:",ACTIONCHGPASS)
+ DIALOGCLOSE
+END [PROCEDURE]
+
+PROCEDURE CHGPASSDIALOG
+[* Dialog that prompts user for new HB password and password
+** probation preference.
+*]
+ DIALOGSTART("Banno Enrollment - New Password",200%,0)
+ DIALOGINTROTEXT("Changing Password for User "+PREFUSER(SELECTEDPREF))
+ DIALOGPROMPTCHAR("Enter New Password:",MAXPASSLENGTH,"")
+ DIALOGPROMPTYESNO("Change Password at next login?",FALSE)
+ DIALOGDISPLAY
+ NEWPASSWORD=ENTERCHARACTER("Enter New Password:",MAXPASSLENGTH,"")
+ NEWPASSPROB=ENTERYESNO("Change Password at next login?","0")
+ DIALOGCLOSE
+END [PROCEDURE]
+
+PROCEDURE CHGPASSFM
+[* File maintenance and error handling for password change.
+*]
+ IF NEWPASSPROB="Y" THEN
+  NEWPASSPROBNUM=1
+ ELSE
+  NEWPASSPROBNUM=0
+
```

Comment:
I think you will need something like in the NT Enrollment PowerOn here, to deal with Hashed passwords.
PASSWORDHASHCODE=GETDATANUMBER(GETPARAMPASSWORDHASHCODE,0)
...
 IF PASSWORDHASHCODE=TRUE THEN
  HASHEDHBPASSWORD=PASSWORDHASH(HBPASSWORD)
 ELSE
  HASHEDHBPASSWORD=HBPASSWORD
...

I also suggest back in the previous procedure doing the password validation and feedback to the user that is in the NT Enrollment PowerOn (but this one is more of a suggestion)

Reply:
This is complete. I have added the code to the CHGPASSFM and CREATEEDITFM procedures.